tests/validator/validator.cpp

```diff
@@ -1065,3 +1063,726 @@ TEST(Validator, validMathCnElements)
+    u3->addUnit("u2", 2.0); // standard, exponent.
+
+    // m^6
+    libcellml::UnitsPtr u4 = std::make_shared<libcellml::Units>();
+    u4->setName("u4");
+    u4->addUnit("u3", 15, 1.0); // standard, prefix, exponent.
+
+    libcellml::UnitsPtr u5 = std::make_shared<libcellml::Units>();
+    u5->setName("u5");
+    u5->addUnit("metre", 6.0); // standard, exponent.
+
+    // u4 = u5: Units will be equivalent, testing that prefix, multiplier, and exponent validation is correct.
+    // TODO: see issue in specification: https://github.com/cellml/cellml-specification/issues/19.
+    v1->setUnits(u4);
+    v2->setUnits(u5);
+
+    comp1->setName("component1");
+    comp1->addVariable(v1);
+
+    comp2->setName("component2");
+    comp2->addVariable(v2);
+
+    model->setName("model");
+    model->addComponent(comp1);
+    model->addComponent(comp2);
+
+    model->addUnits(u1);
+    model->addUnits(u2);
+    model->addUnits(u3);
+    model->addUnits(u4);
+    model->addUnits(u5);
+
+    libcellml::Variable::addEquivalence(v1, v2);
+
+    validator.validateModel(model);
+    EXPECT_EQ(size_t(0), validator.errorCount());
+}
+
+TEST(Validator, unitUserCreatedBaseUnits)
+{
+    std::vector<std::string> expectedErrors = {
+        "Variable 'v1' has units of 'bushell_of_apples' and an equivalent variable 'v2' with non-matching units of 'bunch_of_bananas'. The mismatch is: apple^10, banana^-5, ",
+        "Variable 'v4' has units of 'gram' and an equivalent variable 'v3' with non-matching units of 'litre'. The mismatch is: kilogram^1, metre^-3, ",
+        "Variable 'v7' has units of 'apple' and an equivalent variable 'v8' with non-matching units of 'banana'. The mismatch is: apple^1, banana^-1, ",
+        "Variable 'v2' has units of 'bunch_of_bananas' and an equivalent variable 'v1' with non-matching units of 'bushell_of_apples'. The mismatch is: apple^-10, banana^5, ",
+        "Variable 'v5' has units of 'metre' and an equivalent variable 'v6' with non-matching units of 'second'. The mismatch is: metre^1, second^-1, ",
+        "Variable 'v8' has units of 'banana' and an equivalent variable 'v7' with non-matching units of 'apple'. The mismatch is: apple^-1, banana^1, ",
+        "Variable 'v3' has units of 'litre' and an equivalent variable 'v4' with non-matching units of 'gram'. The mismatch is: kilogram^-1, metre^3, ",
+        // "Variable 'v3' has units of 'litre' and an equivalent variable 'v9' with non-matching units of 'big_barrel'. The mismatch is: multiplication factor of 10^-3, ", // NOTE: removed until multiplication checking is turned on.
+        "Variable 'v6' has units of 'second' and an equivalent variable 'v5' with non-matching units of 'metre'. The mismatch is: metre^-1, second^1, ",
+        // "Variable 'v9' has units of 'big_barrel' and an equivalent variable 'v3' with non-matching units of 'litre'. The mismatch is: multiplication factor of 10^3, ", // NOTE: removed until multiplication checking is turned on.
```

Comment:
Why are there still a couple of expected errors that are commented out? Am I to understand that this PR is not complete in fact? If so, then I guess that commit 14ea753 should be reverted?...

Also, like in another test, the expected errors are effectively duplicated, which is more confusing than anything. I would really have only one expected error between say `v1` and `v2` rather than one between `v1` and `v2` and another between `v2` and `v1`.

Reply:
It's just as the comments say ... removed until multiplication checking is turned on.

Reply:
Sure, but... why isn't this part of this PR?

Reply:
Ok, I imagine this is because we can only generate errors at this stage, as opposed to issues (which may be of either error, warning or hint type). Fair enough, but this means that this piece of commented out code should be removed. Then, if there isn't already one, a GitHub issue should be created to explain the additional work that will need to be done once issues are available.